selftests/functional/test_basic.py

```diff
@@ -391,6 +404,26 @@ def test_fail_on_exception(self):
                                                                 result))
         self.assertIn('"status": "FAIL"', result.stdout)
 
+    def test_exception_not_in_path(self):
+        os.chdir(basedir)
+        os.mkdir(os.path.join(self.tmpdir, "shared_lib"))
+        mylib = script.Script(os.path.join(self.tmpdir, "shared_lib",
+                                           "mylib.py"),
+                              "from avocado import TestCancel\n\n"
+                              "class CancelExc(TestCancel):\n"
```

Comment:
3 lines actually. Bigger than 5 (out of 12) of the current global variables we have for the same purpose, all of them used only in one place. Besides consistency, helps readability. While reading your code, I wondered where would the `CancelExc` in the main script come from. It took me some extra seconds to figure out. Anyway, I'm not against it, so, since you're on it, could you please clean up the other global variables (separate commit, same PR)?